globus_action_provider_tools/flask/apt_blueprint.py

```diff
@@ -217,22 +177,31 @@ def wrapper() -> ViewReturn:
         print(f'Registered action run plugin "{func.__name__}"')
         return wrapper
 
-    @methdispatch
-    def _generic_action_status(self, *args, **kwargs):
-        raise NotImplemented(
-            f"No action_status function registered to handle type {type(args[0])}"
-        )
+    @overload
+    def action_status(self, func: Callable[[str, AuthState], ActionStatusReturn]):
+        """
+        Using these stubs w/ @overload tells mypy that the actual implementation
+        for action_status can accept a str or ActionStatus as the first arg type
+        NOTE: typing_extensions.Protocol would be better if not for it's poor
+        error messages
+        """
+        ...
+
+    @overload
+    def action_status(
+        self, func: Callable[[ActionStatus, AuthState], ActionStatusReturn]
+    ):
+        """
+        Using these stubs w/ @overload tells mypy that the actual implementation
+        for action_status can accept a str or ActionStatus as the first arg type
+        """
+        ...
 
-    def action_status(self, func: ActionStatusType):
+    def action_status(self, func) -> None:
         """
         Decorates a function to be run as an Action Provider's status endpoint.
         """
-        _, first_param = list(inspect.signature(func).parameters.items())[0]
-        if first_param.annotation == first_param.empty:
-            # There's no annotation available, assume the function handles str
-            self._generic_action_status.register(str, func)
-        else:
-            self._generic_action_status.register(first_param.annotation, func)
+        self._action_status: ActionStatusType = func
```

Comment:
Any preference for initializing these sorts of values to `None` in ``__init__`` and then doing an `is None` check rather than `except AttributeError`? I guess the attribute error is considered pythonic under forgiveness/not permission, but I float the alternative. I'm fine either way.